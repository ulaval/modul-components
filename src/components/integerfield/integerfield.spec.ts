import { mount, Wrapper } from '@vue/test-utils';
import Vue from 'vue';
import { renderComponent } from '../../../tests/helpers/render';
import uuid from '../../utils/uuid/uuid';
import IntegerfieldPlugin, { MIntegerfield } from './integerfield';


jest.mock('../../utils/uuid/uuid');
(uuid.generate as jest.Mock).mockReturnValue('uuid');

describe('MTextfield', () => {
    beforeEach(() => {
        Vue.use(IntegerfieldPlugin);
    });
    it('should render correctly', () => {
        const component: Wrapper<MIntegerfield> = mount(MIntegerfield, {
            localVue: Vue
        });

        return expect(renderComponent(component.vm)).resolves.toMatchSnapshot();
    });
});
