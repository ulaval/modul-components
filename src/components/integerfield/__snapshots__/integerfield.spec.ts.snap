// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MTextfield should render correctly 1`] = `
<div class="m-integerfield" style="width:100%;max-width:288px;">
  <div class="m-input-style m--is-tag-default">
    <div class="m-input-style__content" style="margin-top:10px;"><span class="m-input-style__label" style="z-index:-1;"><span class="m-input-style__text"></span></span>
      <div class="m-input-style__body">
        <div class="m-input-style__prefix"></div>
        <div class="m-input-style__left-content"> <input id="mIntegerfield-uuid" inputmode="decimal" class="m-integerfield__input"> </div>
        <div class="m-input-style__right-content"></div>
      </div>
    </div>
  </div>
  <div class="m-integerfield__validation">
    <div class="m-validation-message m-integerfield__validation__message"> </div>
  </div>
</div>
`;
